db/db_impl/db_impl_compaction_flush.cc

```diff
@@ -252,23 +259,25 @@ Status DBImpl::FlushMemTableToOutputFile(
       // be pessimistic and try write to a new MANIFEST.
       // TODO: distinguish between MANIFEST write and CURRENT renaming
       if (!versions_->io_status().ok()) {
-        if (total_log_size_ > 0) {
+        if (total_log_size_ > 0 || !log_io_s.ok()) {
```

Comment:
Not clear why we need to check ```log_io_s``` here and on line 270. If ```SyncClosedLogs``` failed, we shouldn't see SST or manifest write error right, since those wouldn't even be attempted?

Reply:
Is it better to add an assert(log_io_s.ok()) here inside the if to make sure the logic is correct?

Reply:
Yes, I think so.